food-truck-api/src/main/java/food/truck/api/reviews_and_subscriptions/ReviewService.java

```diff
@@ -55,34 +57,34 @@ public Review saveReview(Long userId, Truck t, int score, int costRating, String
     public boolean deleteReviewByUser(User u, long truckId) {
         Review r = reviewRepository.findByUserIdAndTruckId(u.getId(), truckId).stream().findFirst().orElse(null);
         if (r == null)
-            return false; // TODO make return a 404
-        else if (!r.getUserId().equals(u.getId()))
-            return false; // TODO make return a 403
+            return false;
 
         // Delete review
         reviewRepository.delete(r);
         updateAverageRating(r);
         return true;
     }
 
-    public Review findReviewForUser(Long userId, Long truckId) {
-        return reviewRepository.findByUserIdAndTruckId(userId, truckId).stream().findFirst().orElse(null);
+    public Optional<Review> findReviewForUser(Long userId, Long truckId) {
+        return reviewRepository.findByUserIdAndTruckId(userId, truckId).stream().findFirst();
     }
 
-    /**
-     * I hate this solution, but it didn't work out any other way...
-     */
     private void updateAverageRating(Review review) {
         var truck = review.getTruck();
-        var coll = reviewRepository.findByTruckId(review.getId());
+        var coll = reviewRepository.findByTruckId(truck.getId());
 
         // Get new average cost and star ratings
-        long cost = coll.parallelStream()
-                .collect(Collectors.averagingInt(r -> r.getCostRating()))
-                .intValue();
-        long star = coll.parallelStream()
-                .collect(Collectors.averagingInt(r -> r.getCostRating()))
-                .intValue();
+        Long cost = null;
+        Long star = null;
+
+        if (coll.size() != 0) {
+            cost = coll.stream()
+                    .collect(Collectors.averagingInt(Review::getCostRating))
+                    .longValue();
+            star = coll.stream()
+                    .collect(Collectors.averagingInt(Review::getStarRating))
+                    .longValue();
```

Comment:
For the average rating, I think it would make sense for it to be a double so a 4.95-star truck is treated better than a 4.01-star

Reply:
I can look into it, just not sure how many hoops I'd need to jump through with the rating selector components on the frontend.

Reply:
Added it